metrics/src/main/scala/org/broadinstitute/dsde/rawls/metrics/WorkbenchInstrumented.scala

```diff
@@ -36,10 +38,19 @@ trait WorkbenchInstrumented extends DefaultInstrumented {
     *
     * Note the above will only compile if there are [[Expansion]] instances for the types passed to the expand method.
     */
-  protected class ExpandedMetricBuilder private (m: String = "") {
+  protected class ExpandedMetricBuilder private (m: String = "", _transient: Boolean = false) {
     def expand[A: Expansion](key: String, a: A): ExpandedMetricBuilder = {
       new ExpandedMetricBuilder(
-        (if (m == "") m else m + ".") + implicitly[Expansion[A]].makeNameWithKey(key, a))
+        (if (m == "") m else m + ".") + implicitly[Expansion[A]].makeNameWithKey(key, a), _transient)
+    }
+
+    /**
+      * Marks a metric as "transient". Transient metrics will automatically be deleted in Hosted
+      * Graphite if they haven't received an update in X amount of time. It's usually good to set
```

Comment:
I tested this on the Hosted Graphite end -- I set an expiration of 5 days for the metric prefix "prod.firecloud.rawls.workspace" and it deleted a ton of old submission/workflow metrics. Once this is released we can set the prefix to be "prod.firecloud.rawls.transient".